test_data/asm/sqrt.asm

```diff
@@ -28,8 +28,9 @@ machine Sqrt with
             sqrt_rec((y + x / y) / 2, x)
         };
 
-    col witness y(i) query Query::Hint(sqrt_hint(std::prover::eval(x)));
-    
+    col witness y;
+    query |i| std::prover::provide_value(y, i, sqrt_hint(std::prover::eval(x)));
```

Comment:
Are both syntaxes available in the future, i.e. defining `y(i)` as a fixed column via query function vs declaring `y` as a column and `provide_value` to it?

Can you do `col fixed y` and `query |i| provide_value(i)`, i.e. does `provide_value` work for fixed columns as well?

Reply:
I wanted to deprecate the old syntax because it is weird and requires the std::prelude::Query enum. Calling provide_value for fixed columns should result in a runtime error.